micrometer-core/src/main/java/io/micrometer/core/instrument/binder/mongodb/MongoMetricsCommandListener.java

```diff
@@ -56,32 +82,123 @@ public MongoMetricsCommandListener(MeterRegistry registry) {
      * @param registry meter registry
      * @param tagsProvider provides tags to be associated with metrics for the given Mongo command
      * @since 1.7.0
+     * @deprecated use {@link MongoMetricsCommandListener.Builder} instead
      */
+    @Deprecated
     public MongoMetricsCommandListener(MeterRegistry registry, MongoCommandTagsProvider tagsProvider) {
-        this.registry = registry;
-        this.tagsProvider = tagsProvider;
+        this(MongoMetricsCommandListener.builder(registry).tagsProvider(tagsProvider));
+    }
+
+    private MongoMetricsCommandListener(Builder builder) {
+        this.registry = builder.registry;
+        this.timerMetricName = builder.timerMetricName;
+        this.timerMetricDesc = builder.timerMetricDescription;
+        this.tagsProvider = builder.tagsProvider;
+        this.includeCollectionNameTag = builder.includeCollectionNameTag;
+        if (builder.collectionNamesCacheMaxSize < 1) {
+            throw new IllegalArgumentException("collectionNamesCacheMaxSize must be set to a positive value");
+        }
+        this.collectionNamesCacheMaxSize = builder.collectionNamesCacheMaxSize;
+        this.collectionNamesCacheOverflowErrorTracker = builder.collectionNamesCacheOverflowErrorLogInterval < 1 ? null : new EveryNthOperation(builder.collectionNamesCacheOverflowErrorLogInterval);
     }
 
     @Override
-    public void commandStarted(CommandStartedEvent commandStartedEvent) {
+    public void commandStarted(CommandStartedEvent event) {
+        if (!includeCollectionNameTag) {
```

Comment:
I will revisit.

Reply:
I put this in originally as a way for a consumer to basically disable the collection name aspect in case there were issues related to the cache etc.. Somewhat of an emergency lever. I am ok w/ the impl as it is now though - we have been running a clone of this in PROD now for 6 months. I will remove the safeguard. If there is a bug - we will instead fix it quickly.